awx/main/utils/common.py

```diff
@@ -1156,7 +1156,7 @@ def create_partition(tblname, start=None):
                     f'FOR VALUES FROM (\'{start_timestamp}\') TO (\'{end_timestamp}\');'
                 )
     except (ProgrammingError, IntegrityError) as e:
-        if 'already exists' in str(e):
+        if f'{tblname}_{partition_label}, 2200)' in str(e):
```

Comment:
Excuse my lack of understanding. but what is this 'magic number' 2200?
thank you

Reply:
Magic number 2200 is PSQL error code when it's hitting duplicate key (at least what I saw from the error on the internet) - https://www.postgresql.org/docs/current/errcodes-appendix.html

Reply:
for example here: https://www.postgresql.org/message-id/4F9541D3.1000202@beccati.com

Reply:
Is there a constant that represents this code we can use instead? If not lets add a comment

Reply:
Not just that, but I strongly believe we need to get away from doing stringification of exceptions and then scanning the result for substrings.

Reply:
Django's db exception wrapper keeps the original psycopg 3.x exception, which has the code as an attribute:

```
In [11]: try:
    ...:     with connection.cursor() as cursor:
    ...:         cursor.execute("select 1 / 0;")
    ...: except DatabaseError as e:
    ...:     print(e.__context__.sqlstate)
    ...: 
22012
```